sktime/forecasting/ttm.py

```diff
@@ -373,6 +429,20 @@ def _fit(self, y, X, fh):
             y_train = y
             y_eval = None
 
+        # Validation check for proper error message during training
+        val_length = len(y_eval)
```

Comment:
`y_eval` can be `None` in which case, this raises error
```suggestion
        min_length = len(y_train) if y_eval is None else min(len(y_train), len(y_eval))
```